### PR TITLE
Airflow pipeline creation updates

### PR DESCRIPTION
**Pull request by rggelles**
This is a whole bunch of things that needed to work in concert so I'm sorry for the huge review here. I tried to split it out some along the different commits but I'm not sure how successful I was.

Basically, this set of updates should convert PARAT updating over from a set of individual updates to a coherent airflow pipeline, while also implementing all the needed changes for the great update.

Everything appears to be working (the pipeline has completed) although I will need to migrate from airflow dev to production once this is reviewed.

### CODE REVIEW

File: company_linkage/parat_data_dag.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+            },
+        )
+        curr >> next_tab
+        curr = next_tab
+    curr >> wait_for_initial_tables
+
+    # run aggregate_organizations python and load to GCS
+    aggregated_table = "aggregated_organizations"
+
+    aggregate_organizations = PythonOperator(
```
Comment (jmelot) [inline company_linkage/parat_data_dag.py:133]: This doesn't have any significant disc or memory requirements, right? If it does we should move it to a GKE operator since we'll have to do that anyway when we move this dag to cloud composer 2. If not it's ok to leave it as a python operator.

File: company_linkage/README.md [hunk 2]
```diff
@@ -16,37 +16,47 @@ run some of this code as-is.
...
-27. [adding_all_paper_counts.sql](sql/adding_all_paper_counts.sql)
-28. [creating_workforce_visualization_data.sql](sql/creating_workforce_visualization_data.sql)
-29. [adding_ai_jobs_to_workforce_visualization.sql](sql/adding_ai_jobs_to_workforce_visualization.sql)
-31. [omit_by_rule.sql](sql/omit_by_rule.sql)
-32. [omit_by_rule_papers.sql](sql/omit_by_rule_papers.sql)
-33. [omit_by_rule_patents.sql](sql/omit_by_rule_patents.sql)
-34. [omit_by_rule_workforce.sql](sql/omit_by_rule_workforce.sql)
-35. [adding_crunchbase_company_metadata.sql](sql/adding_crunchbase_company_metadata.sql)
\ No newline at end of file
+1. [organizations.sql](sql/organizations.sql)
```
[the remark this answers is not in the document]
Comment (rggelles) [reply company_linkage/README.md:19]: That seems reasonable to me; I can remove it! I used it as I was making the DAG to reorder everything correctly. But that's done now.

File: company_linkage/parat_data_dag.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+
+    join_tables = []
+    for table in ["alias", "grid", "ids", "linkedin", "market", "organizations", "parent", "permid"]:
+
+        # Grab all the data and write it to unseen_en_corpus
+        join_table = BigQueryInsertJobOperator(
+            task_id=f"join_{table}",
+            configuration={
+                "query": {
+                    "query": f"select distinct * from {initial_dataset}.{table}_preannotation UNION DISTINCT "
```
[the remark this answers is not in the document]
Comment (rggelles) [reply company_linkage/parat_data_dag.py:80]: I don't think so? We're very careful about only adding new companies to preannotation, not validate, and there's no duplicates between them from the start.

The only way I can think of duplicates being introduced is if one company acquires another across tables, but we handle that at the aggregate level, not the initial level, so it shouldn't affect things here. I suppose if a company were acquired and its pages and infomation disappeared entirely so it needed to be replaced, which has happened, we could have an issue, but whenever we replace a company during annotation we check whether the company we're replacing it with exists in either list as part of that process.

tl;dr; I think this should be handled through our annotation process

File: company_linkage/parat_data_dag.py (hunk 1 above)
Comment (rggelles) [reply company_linkage/parat_data_dag.py:133]: Nope, extremely limited disk and memory requirements; that's why it's in a Python operator when most other things were using GKE. I can switch it over if you want though.

File: company_linkage/sql/workforce_visualization_data.sql [hunk 4]
```diff
@@ -0,0 +1,21 @@
+  -- Selecting the companies we want to leave out
+WITH
```
[the remark this answers is not in the document]
Comment (rggelles) [reply company_linkage/sql/workforce_visualization_data.sql:2]: Ah! I can see how the naming convention here would be confusing. Perhaps I can clear it up with comments? Basically, what is going on here, though, is:

1. In visualization_data, I am running a rule-based omission, and eliminating any orgs, and their corresponding CSET_ids, that break my set of rules
2. I am saving the *qualifying* (not omitted) results in `visualization_data_omit_by_rule`. That is, that table is the result of running my rule-based omission, but contains the non-omitted data.
3. Then, for later tables like this one, I'm checking all my CSET ids against that table to see which ids aren't in my "qualified orgs" table, and omitting any ids that aren't qualified.

I tried to clarify this by naming it "omit by rule" rather than, say "omitted" or "excluded" or something. But I admit that clarification was probably insufficient.

File: company_linkage/parat_data_dag.py (hunk 1 above)
Comment (jmelot) [reply company_linkage/parat_data_dag.py:133]: No that's fine, just double-checking

**Review by jmelot — APPROVED** [review]
👍
